R/estimationdata.R

```diff
@@ -163,11 +166,30 @@ EstimationData <- function(formula = NULL,
     if (error.if.insufficient.obs && n.estimation < length(variable.names))
         stop(gettextf("There are fewer observations (%d)%s(%d)", n.estimation,
                       " than there are variables ", length(variable.names)))
-
     description <- SampleDescription(n.total, n.subset, n.estimation,
                                      Labels(subset), weighted, weight.label, missing, imputation.label, m,
-                                     if(HasOutcome(formula)) "predictor" else "")
-
+                                     if(HasOutcome(formula)) "predictor" else "",
+                                     dummy.adjusted = dummy.adjusted)
+    # Add statements about removing cases with missing outcomes and/or all predictors missing
+    if (missing == "Dummy variable adjustment")
+    {
+        outcome.name <- OutcomeName(formula)
+        missing.outcomes <- is.na(data[[outcome.name]])
+        outcome.index <- which(names(data) == outcome.name)
+        if (ncol(data) == 2)
+            all.predictors.missing <- rep(FALSE, nrow(data))
+        else
+            all.predictors.missing <- apply(data[-outcome.index], 1, function(x) all(is.na(x)))
+        if (any(missing.outcomes | all.predictors.missing))
+        {
+            missing.outcomes <- if (any(missing.outcomes)) "an outcome variable" else NULL
+            all.predictors.missing <- if (any(all.predictors.missing)) "all predictor variables" else NULL
+            description <- paste0(description, paste0(" cases missing ",
+                                                      paste0(c(missing.outcomes, all.predictors.missing),
+                                                             collapse = " or missing "),
```

Comment:
Is the " or missing " case covered by unit tests?

Reply:
It wasn't, I missed that one. Thanks for picking it up. Added a unit test for that case in the latest commit https://github.com/Displayr/flipData/pull/5/commits/6c6ca284c6b850b532544fbe7fb893ea4b37df0a